File: ios/Tests/LiveUpdateProviderTests/MetadataSyncResultTests.swift
import XCTest
@testable import LiveUpdateProvider

final class MetadataSyncResultTests: XCTestCase {
    func testDefaultMetadataSyncResultCarriesMetadata() throws {
        let result: any MetadataSyncResult = DefaultMetadataSyncResult(metadata: ["version": "1.0.0"])

        let version = try XCTUnwrap(result.metadata?["version"] as? String)
        XCTAssertEqual(version, "1.0.0")
    }

    func testDefaultMetadataSyncResultSupportsMissingMetadata() {
        let result: any MetadataSyncResult = DefaultMetadataSyncResult()

        XCTAssertNil(result.metadata)
    }
}

File: android/live-update-provider/src/test/kotlin/io/ionic/liveupdateprovider/MetadataSyncResultTests.kt
package io.ionic.liveupdateprovider

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test

class MetadataSyncResultTests {
    @Test
    fun `default metadata sync result carries metadata`() {
        val result: MetadataSyncResult = DefaultMetadataSyncResult(mapOf("version" to "1.0.0"))

        assertEquals("1.0.0", result.metadata?.get("version"))
    }

    @Test
    fun `default metadata sync result supports missing metadata`() {
        val result: MetadataSyncResult = DefaultMetadataSyncResult()

        assertNull(result.metadata)
    }
}
